### ISSUE TITLE
Add Raven/Sentry logger adapter

### ISSUE THREAD

**Issue by njam** [cargomedia/cm#1860]
Depends on #1926

URL: https://github.com/getsentry/raven-php
Interfaces: http://sentry.readthedocs.org/en/stable/developer/interfaces/index.html

**Comment by vogdb** [comment]
```php

function translateSeverityToLevel($severity) {
    $map = [
        CM_Exception::WARN  => Raven_Client::WARN,
        CM_Exception::ERROR => Raven_Client::ERROR,
        CM_Exception::FATAL => Raven_Client::FATAL,
    ];
    return $map[$severity];
}

function getClient() {
    $client = new Raven_Client(
        'Place DSN here',
        [
            'name' => CM_Util::getFqdn(),
            'tags' => ['foo2' => 'bar']
        ]
    );
    $client->user_context([
        "is_authenticated" => false,
//        "id"               => "unique_id",
//        "username"         => "foo",
//        "email"            => "foo@example.com",
    ]);
    return $client;
}

function sendException(Exception $exception) {
    $client = getClient();
    $options = [];

    if ($exception instanceof CM_Exception) {
        $options['level'] = translateSeverityToLevel($exception->getSeverity());
        $options['extra'] = $exception->getMetaInfo();
    }

    $event_id = $client->getIdent($client->captureException($exception, $options));
    if ($client->getLastError() !== null) {
        printf('There was an error sending the event to Sentry: %s', $client->getLastError());
    }
}

sendException(new CM_Exception('Samurai!', ['foo' => 'bar']));
//$error_handler = new Raven_ErrorHandler($client);
//$error_handler->registerExceptionHandler();
//$error_handler->registerErrorHandler();
//$error_handler->registerShutdownFunction();

```

**Comment by njam** [comment]
```php
LogManager::logException($exception);
 foreach($this->getProvider() as $provider) {
  $provider->logException($exception)
 }

LogManager_Provider_Database:
 logException($exception):
  Paging_Log_Something:add();

LogManager_Provider_Sentry:
 logException($exception):
  Raven_Client::captureException($exception)
```

**Comment by njam** [comment]
@vogdb can you track 1 error with their JS code so we can see how it looks?

**Comment by vogdb** [comment]
done

**Comment by njam** [comment]
What they show for JS:
- Stack trace
- Request:
 - URL
 - Query
 - Headers
- User:
 - IP Address
- Other Tags
 - Browser ("Chrome 44.0")
 - Device ("Other")
 - OS ("Mac OS X 10.10.3")

**Comment by njam** [comment]
Closing in favor of https://github.com/cargomedia/cm/issues/2117
